HLTrigger/Muon/src/HLTScoutingMuonProducer.cc

```diff
@@ -83,36 +86,121 @@ void HLTScoutingMuonProducer::produce(edm::StreamID sid, edm::Event & iEvent,
+	
+	// Get the muons associated with the tracks
+	int iFoundRefs = 0;
+	for (reco::RecoChargedCandidateCollection::const_iterator cand=ChargedCandidateCollection->begin(); cand!=ChargedCandidateCollection->end(); cand++) {
+	  reco::TrackRef tkRef = cand->get<reco::TrackRef>();
+	  if(tkRef == vertextkRef1) {ivtxMuPair.first= (*cand); iFoundRefs++ ;}
+	  if(tkRef == vertextkRef2) {ivtxMuPair.second= (*cand); iFoundRefs++ ;}
+	}
+	if (iFoundRefs<2) continue;
+	vtxMuPair.push_back(ivtxMuPair);
+	
+	dispVertices->emplace_back(
+				   dispvtx.x(), dispvtx.y(), dispvtx.z(), 
+				   dispvtx.zError(), dispvtx.xError(), 
+				   dispvtx.yError(), dispvtx.tracksSize(), 
+				   dispvtx.chi2(), dispvtx.ndof()
+				   );
+	
+      }
+    }
+    
     // Produce muons
+    std::vector<int> vtxInd;
+    vtxInd.clear();
+    float minDR2=1e-06;
     int index = 0;
     for (auto &muon : *ChargedCandidateCollection) {
-        reco::RecoChargedCandidateRef muonRef = getRef(ChargedCandidateCollection, index);
-        ++index;
-        if (muonRef.isNull() || !muonRef.isAvailable())
-            continue;
-
-        reco::TrackRef track = muon.track();
-        if (track.isNull() || !track.isAvailable())
-            continue;
-
-        if (muon.pt() < muonPtCut)
-            continue;
-        if (fabs(muon.eta()) > muonEtaCut)
-            continue;
-
-        outMuons->emplace_back(muon.pt(), muon.eta(), muon.phi(),  muon.mass(),
-                               (*EcalPFClusterIsoMap)[muonRef], (*HcalPFClusterIsoMap)[muonRef],
-                               (*TrackIsoMap)[muonRef], track->chi2(), track->ndof(),
-                               track->charge(), track->dxy(), track->dz(),
-                               track->hitPattern().numberOfValidMuonHits(),
-                               track->hitPattern().numberOfValidPixelHits(),
-                               0, // nMatchedStations
-                               track->hitPattern().trackerLayersWithMeasurement(),
-                               2); // Global muon
+      reco::RecoChargedCandidateRef muonRef = getRef(ChargedCandidateCollection, index);
+      ++index;
+      if (muonRef.isNull() || !muonRef.isAvailable())
+	continue;
+      
+      reco::TrackRef track = muon.track();
+      if (track.isNull() || !track.isAvailable())
+	continue;
+      
+      if (muon.pt() < muonPtCut)
+	continue;
+      if (fabs(muon.eta()) > muonEtaCut)
+	continue;
+      
+      double ecalisopf=-99.0;
+      if  ( !EcalPFClusterIsoMap.isValid() ) ecalisopf = -1.0 ;
+      else ecalisopf = (*EcalPFClusterIsoMap)[muonRef]; 
+      
+      double hcalisopf=-99.0;
+      if  ( !HcalPFClusterIsoMap.isValid() ) hcalisopf = -1.0 ;
+      else hcalisopf = (*HcalPFClusterIsoMap)[muonRef]; 
+      
+      for (unsigned int i=0; i<vtxMuPair.size(); i++) {
+	float dr2_1 = reco::deltaR2( ((vtxMuPair.at(i)).first),muon );
+	float dr2_2 = reco::deltaR2( ((vtxMuPair.at(i)).second),muon );
```

Comment:
There is no need to use `at()` here since you know you are within the allowed indicies. Better to use `vtxMuPair[i]` which is faster.